src/versions/drafts/buildDraftMergeAggregate.ts

```diff
@@ -0,0 +1,79 @@
+import { PipelineStage } from 'mongoose';
+import { SanitizedCollectionConfig } from '../../collections/config/types';
+
+type Args = {
+  config: SanitizedCollectionConfig
+  query: Record<string, unknown>
+}
+
+export const buildDraftMergeAggregate = ({ config, query }: Args): PipelineStage[] => [
+  // Merge in versions by parent ID
+  {
+    $lookup: {
+      from: `_${config.slug}_versions`,
+      as: '_versions',
+      localField: '_id',
+      foreignField: 'parent',
+    },
+  },
+
+  // Add a new field
+  // with the last (newest) version returned
+  {
+    $addFields: {
+      _version: {
+        $arrayElemAt: ['$_versions', -1],
+      },
+    },
+  },
+
+  // If newer version exists,
+  // merge the version into the root and
+  // replace the updatedAt
+  // Otherwise, do nothing to the root
+  {
+    $replaceRoot: {
+      newRoot: {
+        $cond: {
+          if: {
+            $and: [
+              {
+                $ne: ['$_version', undefined],
+              },
+              {
+                $gt: ['$_version.updatedAt', '$updatedAt'],
+              },
+            ],
+          },
+          then: {
+            $mergeObjects: [
+              '$$ROOT',
+              {
+                $mergeObjects: [
+                  '$_version.version',
+                  {
+                    updatedAt: '$_version.updatedAt',
```

Comment:
I had errors when I didn't extract the _id out the main document.  
Also the createdAt of the main document should be kept?

```suggestion
                    updatedAt: '$_version.updatedAt',
                    _id: '$_id',
                    createdAt: '$createdAt',
```